openrtb_ext/imp_undertone.go

```diff
@@ -4,3 +4,7 @@ type ExtImpUndertone struct {
 	PublisherID int `json:"publisherId"`
 	PlacementID int `json:"placementId"`
 }
+
+type UndertoneImpExt struct {
+	Gpid string `json:"gpid,omitempty"`
+}
```

Comment:
The intent of this file is to capture the bidder params for your adapter. Since GPID (`imp.ext.gpid`) is not an Undertone bidder parameter, this data structure might be better suited for the `adapters/undertone/undertone.go` file.

I also recommend combining it with with the Undertone parameters in a single structure so that you just need to call `json.Unmarshal` once instead of three times - as the Unmarshal call can be an expensive operation.

Something like...

```
type impExt struct {
  Bidder *openrtb_ext.ExtImpUndertone `json:"bidder,omitempty"`
  GPID string `json:"gpid,omitempty"`
}

...

for _, imp := range bidRequest.Imp {
	var ext impExt
	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
		errs = append(errs, getInvalidImpErr(imp.ID, err))
		continue
	}

	if publisherId == 0 {
		publisherId = impExt.Bidder.PublisherID
	}

	imp.TagID = strconv.Itoa(impExt.Bidder.PlacementID)
	imp.Ext = nil

	if ext.GPID != "" {
		ext.Bidder = nil
		extJSON, err := json.Marshal(&ext)
		if err != nil {
			errs = append(errs, getInvalidImpErr(imp.ID, err))
			continue
		}
		imp.Ext = impExtJson
	}
```

Reply:
Thanks, I've pushed a new commit with these changes.

Reply:
Nice. Thank for for accepting the suggesiton.